src/app/nav/navigate-action.js

```diff
@@ -0,0 +1,14 @@
+
+/**
+ * An action creator for in-app page navigation.
+ */
+
+module.exports = function (actionContext, payload, done) {
+  var route = actionContext.router.getRoute(payload.url);
+  if (route) {
+    actionContext.dispatch('navigate', route);
+    done();
+  } else {
+    done({ failed: true });
```

Comment:
For some reason when done() was called on the client specifically with parameters it like... threw an error or something. It complained specifically on the client side when done was called with parameters if there was not an error.